server-spi-private/src/main/java/org/keycloak/models/LockObjectsForModification.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models;
```

Comment:
I placed this class in the same package as KeycloakSession. I'd be happy for a comment if it should be placed somewhere else.

Reply:
As far as I'm concerned it's in right place, @hmlnarik wdyt?

Reply:
The module is correct, I would only put it into [`utils`](https://github.com/keycloak/keycloak/tree/main/server-spi-private/src/main/java/org/keycloak/models/utils) subpackage there.